rskj-core/src/main/java/co/rsk/peg/BridgeStorageProvider.java

```diff
@@ -923,6 +925,51 @@ protected int getReleaseRequestQueueSize() throws IOException {
         return getReleaseRequestQueue().getEntries().size();
     }
 
+    public boolean hasBridgeBtcTxSigHash(Sha256Hash sigHash) {
+        if (!activations.isActive(RSKIP379) || sigHash == null){
+            return false;
+        }
+
+        byte[] data = repository.getStorageBytes(
+            contractAddress,
+            getStorageKeyForBridgeBtcTxSigHash(sigHash)
+        );
+
+        return data != null &&
+           data.length == 1 &&
+           data[0] == TRUE_VALUE;
+    }
+
+    public void setBridgeBtcTxSigHash(Sha256Hash sigHash) {
+        if (!activations.isActive(RSKIP379) || sigHash == null) {
+            return;
+        }
+
+        if (hasBridgeBtcTxSigHash(sigHash)){
+            throw new IllegalStateException(String.format("Given bridge btc tx sigHash %s already exists in the index. Index entries are considered unique.", sigHash));
+        }
+
+        if (bridgeBtcTxSigHashes == null){
+            bridgeBtcTxSigHashes = new HashSet<>();
+        }
+
+        bridgeBtcTxSigHashes.add(sigHash);
+    }
+
+    private void saveBridgeBtcTxSigHashes() {
+        if (!activations.isActive(RSKIP379) || bridgeBtcTxSigHashes == null) {
+            return;
+        }
+
+        bridgeBtcTxSigHashes.forEach(bridgeBtcTxSigHash -> repository.addStorageBytes(
```

Comment:
```suggestion
       log.warn("[saveBridgeBtcTxSigHashes] About to save {} sigh hashes.", bridgeBtcTxSigHashes.size())
        bridgeBtcTxSigHashes.forEach(bridgeBtcTxSigHash -> repository.addStorageBytes(
```
Nice To Have a Log Here Incase We Encounter The Rare Case We Talked About

Reply:
Can you remind me what case it is? Even though I believe no log is needed. There is not even a logger instance currently in the bridge storage provider class. None of the storage provider methods is logging anything now.

Reply:
The Case Where We Have Both a Pegout And Migration Transaction At The Same Time. So We Will Have 2 Hashes Set To `bridgeBtcTxSigHashes`. Its a Rare Case That We Want To Log If It Happens For Debugging Sake.

Reply:
If that happens, it will be fine. I changed the implementation of the index. Now it supports a list of `pegoutTxSigHashes`, so it is capable of storing `n` sigHashes per rsk tx as long as they are different. 

However, the case you mention will only be possible if we store the entries at the moment when peg-outs are created. But we do not pretend to do it at that time. We will do it once the peg-out is fully signed. This is on the `addSignature` method. If in the future the bridge adds the capability to add a signature for more than one peg-out at a time, no prob, we are covered.

So basically, we are fine so far.